### PR TITLE
feat(lnurl): support channelRequest #738

### PR DESCRIPTION
**Pull request by escapedcat**
### Describe the changes you have made in this PR

Support LNURLChannel via connectPeer.  
Currently only supports LND.

### Link this PR to an issue

#738

### Type of change (Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### How has this been tested?

Via devtools

### Checklist

- [ ] My code follows the style guidelines of this project and performed a self-review of my own code
- [ ] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/extension/background-script/actions/lnurl/index.ts [hunk 1]
```diff
@@ -20,10 +21,11 @@ async function lnurl(message: Message) {
 
   switch (lnurlDetails.tag) {
     case "channelRequest":
-      console.error("lnurl-channel is not implemented");
-      // TODO: add support for LNURL channel
-      // connectPeer method in connector
-      return { error: "not implemented" };
+      if ("public" in message) {
```
Comment (bumi) [inline src/extension/background-script/actions/lnurl/index.ts:24]: what is this check about? and why only a `return` in the else case?

File: src/extension/background-script/actions/ln/getBalance.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import state from "../../state";
+
+const getInfo = async (message, sender) => {
```
Comment (bumi) [inline src/extension/background-script/actions/ln/getBalance.js:3]: what is this getBalance.js for? where is it used.
and I think you have a c&p error here => `getInfo`?

**Review by bumi — COMMENTED** [review]
I still think we should call it lnurlchannel and not lnurlopenchannel because the spec is called lnurl channel and it actually also does not open a channel (it connects a peer and does a channel request to a service)

File: src/app/screens/LNURLOpenChannel.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  async function confirm() {
+    try {
+      setErrorMessage("");
+      setLoadingConfirm(true);
+      await api.connectPeer({
+        host,
+        pubkey,
+      });
+      // todo: check success
```
Comment (escapedcat) [inline src/app/screens/LNURLOpenChannel.tsx:39]: This is outdated, yes?

Comment (bumi) [reply src/app/screens/LNURLOpenChannel.tsx:39]: +1 looks like 

though we do not check if the `connectPeer` call was successful?

File: src/extension/background-script/actions/lnurl/index.ts (hunk 1 above)
Comment (bumi) [reply src/extension/background-script/actions/lnurl/index.ts:24]: hmm. now we got another else there. and all the other lnurl options are missing. 
should that then be 

```js 
lnurl(MessageLNURLChannel | MessageLNURLPay | MessageLNURLAuth | ...)
```

and where is that `public` coming from?

File: src/app/screens/LNURLOpenChannel.tsx (hunk 3 above)
Comment (escapedcat) [reply src/app/screens/LNURLOpenChannel.tsx:39]: No, but doesn't this take care of it?
https://github.com/getAlby/lightning-browser-extension/pull/1173/files#diff-75c67e9b6d920ae59f2bb6ec2011595392fa96986711b39bb1e57f36ded8f5ceR21-R28

Comment (bumi) [reply src/app/screens/LNURLOpenChannel.tsx]: ah ok, we have that global `try {` where we catch that error. 🙈

**Pull request merged**